### PR TITLE
Lazy tentative to replace of auto_ptr with unique_ptr and move to c++14

### PR DESCRIPTION
**Pull request by sloriot**
I'm not claiming this is perfect, but at least it compiles and the testsuite (simply running ctest) works.

### CODE REVIEW

**Comment by vmora** [comment]
"au temps pour moi" and thanks a lot @sloriot

**Comment by mhugo** [comment]
@sloriot Thanks !

File: src/PreparedGeometry.cpp [hunk 1]
```diff
@@ -28,8 +28,8 @@ PreparedGeometry::PreparedGeometry() :
 {
 }
 
-PreparedGeometry::PreparedGeometry( std::auto_ptr<Geometry> geometry, srid_t srid ) :
-    _geometry( geometry ),
+PreparedGeometry::PreparedGeometry( std::unique_ptr<Geometry>&& geometry, srid_t srid ) :
+    _geometry( geometry.release() ),
```
Comment (lrineau) [inline src/PreparedGeometry.cpp]: When a function claim ownership of a pointer, the `unique_ptr` is best taken by value:
```c++
PreparedGeometry::PreparedGeometry( std::unique_ptr<Geometry> geometry, srid_t srid ) 
    _geometry( std::move(geometry) ),
```

For a long explanation, you can read https://stackoverflow.com/a/8114913/1728537.

The semantic of `void f(std::unique_ptr<Geometry>&& geometry)` is that `f` *may* take the ownership.

Comment (mhugo) [reply src/PreparedGeometry.cpp]: I agree, passing rvalue references of unique_ptr seems very confusing to me.

**Pull request merged**